i3, i3-gaps: 4.18.3 -> 4.19

marzzzello opened the pull request:
Upstream switched from the autotools build system to the meson build system.
Also the source archive format switched from tar.bz2 to tar.xz
The build now supports application documentation in /share/doc/i3/
https://i3wm.org/downloads/RELEASE-NOTES-4.19.txt

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

New features and fixes

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

marzzzello committed:
i3, i3-gaps: 4.18.3 -> 4.19

Upstream switched from the autotools build system to the meson build system.
Also the source archive format switched from tar.bz2 to tar.xz
The build now supports application documentation in /share/doc/i3/
https://i3wm.org/downloads/RELEASE-NOTES-4.19.txt

marzzzello commented:
i3-gaps and python39Packages.powerline are failing. i3-gaps also switched to the new build system. I already have i3-gaps working with 4.19. Should I do a separate PR how to proceed?
```
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/marcel/.cache/nixpkgs-review/pr-109106-2/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
builder for '/nix/store/j6irxp0p0f26p8al5f1pvy344kk01abx-i3-gaps-4.18.3.drv' failed with exit code 1; last 10 log lines:
  autoreconf: running: /nix/store/var821ibfzbhjk911z4b1h5xhxc3pfl5-autoconf-2.70/bin/autoheader --force
  autoreconf: running: automake --add-missing --copy --force-missing
  Makefile.am:229: warning: '%'-style pattern rules are a GNU make extension
  autoreconf: './config.sub' is updated
  autoreconf: './config.guess' is updated
  autoreconf: Leaving directory '.'
  configuring
  meson flags: --buildtype=plain         --libdir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/lib --libexecdir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/libexec         --bindir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/bin --sbindir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/sbin         --includedir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/include         --mandir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/share/man --infodir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/share/info         --localedir=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3/share/locale         -Dauto_features=enabled         -Dwrap_mode=nodownload         --prefix=/nix/store/41wh68b6sqgy53kql7937yxyaa5y18l2-i3-gaps-4.18.3  
  
  ERROR: Neither directory contains a build file meson.build.
builder for '/nix/store/f8gfdpqmjvb5lbf3wm824jszh6fc2iq1-python3.9-pyuv-1.2.0.drv' failed with exit code 1; last 10 log lines:
  src/pyuv.c:72:5: warning: ‘PyEval_InitThreads’ is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
     72 |     PyEval_InitThreads();
        |     ^~~~~~~~~~~~~~~~~~
  In file included from /nix/store/3g9596lia3mdr753amydqs6ilr4iaf1m-python3-3.9.1/include/python3.9/Python.h:145,
                   from src/pyuv.h:7,
                   from src/pyuv.c:2:
  /nix/store/3g9596lia3mdr753amydqs6ilr4iaf1m-python3-3.9.1/include/python3.9/ceval.h:130:37: note: declared here
    130 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
        |                                     ^~~~~~~~~~~~~~~~~~
  error: command '/nix/store/d2fqghv083a8q7j8sjb2m5q6i840h691-gcc-wrapper-10.2.0/bin/gcc' failed with exit code 1
cannot build derivation '/nix/store/cyijpz2r2kpb1qv0ccamzs6shgjlgji3-python3.9-powerline-2.8.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/fb4dn34ik8bxsb6a1nql54r7rz29dg6l-env.drv': 2 dependencies couldn't be built
[0 built (2 failed), 0.0 MiB DL]
error: build of '/nix/store/fb4dn34ik8bxsb6a1nql54r7rz29dg6l-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/109106
2 packages failed to build:
i3-gaps python39Packages.powerline

20 packages built:
autotiling flashfocus i3 i3-balance-workspace i3-layout-manager i3-resurrect i3-wk-switch i3altlayout polybarFull powerline python37Packages.i3ipc python37Packages.powerline python37Packages.py3status python38Packages.i3ipc python38Packages.py3status python39Packages.i3ipc python39Packages.py3status raiseorlaunch sway-contrib.inactive-windows-transparency xkb-switch-i3
```

SuperSandro2000 commented on pkgs/applications/window-managers/i3/default.nix:
```diff
@@ -1,18 +1,18 @@
   };
 
-  nativeBuildInputs = [ which pkgconfig makeWrapper installShellFiles ];
+  nativeBuildInputs = [ which pkgconfig makeWrapper meson ninja installShellFiles ];
```
Do we still require which and pkg-config?

If so please change pkgconfig to pkg-config.

SuperSandro2000 commented:
> Should I do a separate PR or how to proceed?

You can append it to here.

jonringer commented:
the current i3-gaps just overrides the i3 package, we may want to keep a `i3_4_18` package so that i3 is able to build.

related: https://github.com/Airblader/i3/issues/381

marzzzello commented:
I removed `which` and changed `pkgconfig` to `pkg-config`.
And I added i3-gaps. 
python39Packages.powerline still fails to build.
Any ideas how to fix this? @SuperSandro2000
```
$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/marcel/.cache/nixpkgs-review/pr-109106-4/build.nix
warning: ignoring the user-specified setting 'sandbox', because it is a restricted setting and you are not a trusted user
builder for '/nix/store/f8gfdpqmjvb5lbf3wm824jszh6fc2iq1-python3.9-pyuv-1.2.0.drv' failed with exit code 1; last 10 log lines:
  src/pyuv.c:72:5: warning: ‘PyEval_InitThreads’ is deprecated [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
     72 |     PyEval_InitThreads();
        |     ^~~~~~~~~~~~~~~~~~
  In file included from /nix/store/3g9596lia3mdr753amydqs6ilr4iaf1m-python3-3.9.1/include/python3.9/Python.h:145,
                   from src/pyuv.h:7,
                   from src/pyuv.c:2:
  /nix/store/3g9596lia3mdr753amydqs6ilr4iaf1m-python3-3.9.1/include/python3.9/ceval.h:130:37: note: declared here
    130 | Py_DEPRECATED(3.9) PyAPI_FUNC(void) PyEval_InitThreads(void);
        |                                     ^~~~~~~~~~~~~~~~~~
  error: command '/nix/store/d2fqghv083a8q7j8sjb2m5q6i840h691-gcc-wrapper-10.2.0/bin/gcc' failed with exit code 1
cannot build derivation '/nix/store/s5p8jclks8j34sp395b0640br985xknh-python3.9-powerline-2.8.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/kla3rfr9pczssss5l53milhhq09wdmbk-env.drv': 1 dependencies couldn't be built
[25 built (1 failed), 1 copied (0.2 MiB), 0.0 MiB DL]
error: build of '/nix/store/kla3rfr9pczssss5l53milhhq09wdmbk-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/109106
1 package failed to build:
python39Packages.powerline

21 packages built:
autotiling flashfocus i3 i3-balance-workspace i3-gaps i3-layout-manager i3-resurrect i3-wk-switch i3altlayout polybarFull powerline python37Packages.i3ipc python37Packages.powerline python37Packages.py3status python38Packages.i3ipc python38Packages.py3status python39Packages.i3ipc python39Packages.py3status raiseorlaunch sway-contrib.inactive-windows-transparency xkb-switch-i3
```

jonringer commented:
python39Packages.powerline was failing before, and broken on master, don't worry about that one

jonringer reviewed (approved):
LGTM

```
python39Packages.powerline

21 packages built:
autotiling flashfocus i3 i3-balance-workspace i3-gaps i3-layout-manager i3-resurrect i3-wk-switch i3altlayout polybarFull powerline python37Packages.i3ipc python37Packages.powerline python37Packages.py3status python38Packages.i3ipc python38Packages.py3status python39Packages.i3ipc python39Packages.py3status raiseorlaunch sway-contrib.inactive-windows-transparency xkb-switch-i3
```

jonringer commented:
waiting for ofborg
